src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3693,6 +3720,75 @@ void Lowering::LowerHWIntrinsicToScalar(GenTreeHWIntrinsic* node)
         LowerNode(cast);
     }
 }
+
+//----------------------------------------------------------------------------------------------
+// Lowering::LowerAndOpToResetLowestSetBit: Lowers a tree AND(X, ADD(X, -1) to HWIntrinsic::ResetLowestSetBit
+//
+// Arguments:
+//    andNode - GentreePtr of binOp
+//
+// Return Value:
+//    Returns the replacement node if one is created else nullptr indicating no replacement
+//
+// Assumptions:
+//    andNode is GT_AND and (TYP_INT or TYP_LONG)
+//
+GenTree* Lowering::LowerAndOpToResetLowestSetBit(GenTree* andNode)
+{
+    assert(andNode->OperIs(GT_AND) && varTypeIsIntegral(andNode));
+
+    GenTree* op1 = andNode->gtGetOp1();
+    GenTree* op2 = andNode->gtGetOp2();
+    if (op1->OperIs(GT_LCL_VAR) && op2->OperIs(GT_ADD) && !comp->lvaGetDesc(op1->AsLclVar())->IsAddressExposed())
+    {
+        GenTree* addOp1 = op2->gtGetOp1();
+        GenTree* addOp2 = op2->gtGetOp2();
+        if (addOp2->IsIntegralConst(-1) && addOp1->OperIs(GT_LCL_VAR) &&
+            (addOp1->AsLclVar()->GetLclNum() == op1->AsLclVar()->GetLclNum()) &&
+            ((op1->TypeIs(TYP_LONG) && comp->compOpportunisticallyDependsOn(InstructionSet_BMI1_X64)) ||
+             comp->compOpportunisticallyDependsOn(InstructionSet_BMI1))
+            )
```

Comment:
I would expect you can combine the `)` with the `comp->compOpportunisticallyDependsOn(InstructionSet_BMI1))` now.

Reply:
The coding conventions say:

> There should be parentheses around all unary and binary expressions if they are contained within other expressions. We prefer to over-specify parentheses instead of requiring developers to memorize the complete C++ precedence rules. This is especially true for && and ||, whose precedence relationship is often forgotten.

which seems to be ok with the additional parents around the `((islong && issupportedlong) || issupportedint)`

Reply:
Yes, I didn't mean to drop it, but move it up a line, i. e.:
```
              comp->compOpportunisticallyDependsOn(InstructionSet_BMI1))
            ) 
```
->
```
              comp->compOpportunisticallyDependsOn(InstructionSet_BMI1))) 
```

Reply:
I think the formatter has moved it up.